lib/rules/no-duplicate-id.js

```diff
@@ -0,0 +1,39 @@
+'use strict';
+
+const Rule = require('./base');
+
+const ERROR_MESSAGE = 'ID attribute values must be unique';
+
+module.exports = class NoDuplicateId extends Rule {
+  visitor() {
+    let attrIdStack = [];
+    return {
+      AttrNode(node) {
+        if (!(node.name === 'id')) {
+          return;
+        }
+
+        if (!(node.value.type === 'TextNode')) {
+          return;
+        }
+
+        attrIdStack.push(node.value.chars);
+
+        let hasDuplicates = attrIdStack.some(
```

Comment:
It would be more efficient to use an object (which is basically a hashtable) to keep track of which IDs we have seen instead of an array.

Reply:
I would recommend `new Set()`, but I'm not yet sure I entirely understand what the `.some()` here does 😅

Reply:
Ya, agree. We should swap to `Set` and do something like:

```js
if (seen.has(idValue)) {
  this.log(...)
} else {
  seen.add(idValue);
}
```